Plugins/TGeo/include/Acts/Plugins/TGeo/TGeoCylinderDiscSplitter.hpp

```diff
@@ -0,0 +1,72 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2021 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Geometry/GeometryContext.hpp"
+#include "Acts/Plugins/TGeo/ITGeoDetectorElementSplitter.hpp"
+#include "Acts/Utilities/Logger.hpp"
+
+#include <memory>
+
+class TGeoNode;
+
+namespace Acts {
+
+class TGeoDetectorElement;
+
+/// @brief TGeoCylinderDiscSplitter
+///
+/// Split Cylinder and disks into submodules
+class TGeoCylinderDiscSplitter : public ITGeoDetectorElementSplitter {
+ public:
+  /// Nested configuration struct
+  struct Config {
+    /// Number of segments in phi for a disc
+    int cylinderPhiSegments = -1;
+    /// Number of segments in r for a disk
+    int cylinderLongitudinalSegments = -1;
+
+    /// Number of segments in phi for a disc
+    int discPhiSegments = -1;
+    /// Number of segments in r for a disk
+    int discRadialSegments = -1;
+  };
+
+  /// Constructor
+  ///
+  /// @param cfg the configuration struct
+  /// @param logger the logging object
+  TGeoCylinderDiscSplitter(
+      const Config& cfg,
+      std::unique_ptr<const Acts::Logger> logger = Acts::getDefaultLogger(
+          "TGeoCylinderDiscSplitter", Acts::Logging::INFO));
+
+  virtual ~TGeoCylinderDiscSplitter() = default;
```

Comment:
Shouldn't this better go to the Interface?

Reply:
I think because it's only purely virtual it's fine, did not complain.

Reply:
Ok, maybe I'm mistaken, but is the destrcutor of the splitter called through an interface pointer when the interface has no virtual destructor?

Reply:
I've included it now.